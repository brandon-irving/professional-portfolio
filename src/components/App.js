import React from 'react';
import Navbar from './Navbar';
import ThemeButton from './ThemeButton';
import About from './About';
import Tools from './Tools';

function App() {
  return (
    <div className="App">
      <Navbar />
      <About />
      <Tools />
      <ThemeButton />
    </div>
  );
}

export default App;
