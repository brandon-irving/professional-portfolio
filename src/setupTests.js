import React from 'react';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { createSerializer } from 'enzyme-to-json';
import CustomThemeProvider from 'components/CustomThemeProvider';
import 'icons';

configure({ adapter: new Adapter() });

expect.addSnapshotSerializer(createSerializer({ mode: 'deep' }));

export const withTheme = WrappedComponent => (
  <CustomThemeProvider>{WrappedComponent}</CustomThemeProvider>
);
