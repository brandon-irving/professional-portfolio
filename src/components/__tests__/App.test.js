import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import App from '../App';
import { withTheme } from 'setupTests';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(withTheme(<App />), div);
});

it('matches snapshot', () => {
  const wrapper = shallow(<App />);
  expect(wrapper).toMatchSnapshot();
});
