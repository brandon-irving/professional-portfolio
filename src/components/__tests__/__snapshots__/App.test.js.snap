// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<Fragment>
  <Navbar />
  <styled.div>
    <About />
    <Tools />
    <Work />
    <Testimonials />
    <CallToAction />
    <Footer />
  </styled.div>
  <ThemeButton />
</Fragment>
`;
